include/libtorrent/alert_types.hpp

```diff
@@ -1815,8 +1815,8 @@ namespace libtorrent
 	struct TORRENT_EXPORT add_torrent_alert final : torrent_alert
 	{
 		// internal
-		add_torrent_alert(aux::stack_allocator& alloc, torrent_handle h
-			, add_torrent_params const& p, error_code ec);
+		add_torrent_alert(aux::stack_allocator& alloc, torrent_handle const& h
```

Comment:
with move semantics, I think the preferred way of copying arguments is to take them by value and move them into the member. that way the copy is pushed up to the caller, which can do the same thing (i.e. either push the copy up further, to its caller, or move)